libr/anal/fcn.c

```diff
@@ -1862,12 +1862,21 @@ R_API RAnalFunction *r_anal_fcn_find_name(RAnal *anal, const char *name) {
 	return NULL;
 }
 
+#define MAXBBSIZE (1024*1024*1)
 /* rename RAnalFunctionBB.add() */
-R_API int r_anal_fcn_add_bb(RAnal *anal, RAnalFunction *fcn, ut64 addr, ut64 size, ut64 jump, ut64 fail, int type, RAnalDiff *diff) {
+R_API bool r_anal_fcn_add_bb(RAnal *anal, RAnalFunction *fcn, ut64 addr, ut64 size, ut64 jump, ut64 fail, int type, RAnalDiff *diff) {
 	RAnalBlock *bb = NULL, *bbi;
 	RListIter *iter;
 	bool mid = false;
 	st64 n;
+	if (size == 0) { // empty basic blocks allowed?
+		eprintf ("warning: empty basic block at 0x%08"PFMT64x" is not allowed. pending discussion.\n", addr);
+		return false;
+	}
+	if (size > MAXBBSIZE) {
```

Comment:
why? is 1MB so much that we cannot allocate 1MB of data?

Reply:
its just an arbitrary size because 99.9% of the cases that will be a false positive, so better catch it earlier

Reply:
I think it prevents proper analyses. When packers/obfuscators are used I often find very big basic blocks that are truncated because of limits like this. (e.g.  movfuscator)